### PR TITLE
Fix warning msg print

### PR DESCRIPTION
**Pull request by SunnyLee151064**
### What this PR does / why we need it?
Avoid printing some warning msg as below :
UserWarning: To copy construct from a tensor, it is recommended to use sourceTensor.clone().detach ...
### Does this PR introduce _any_ user-facing change?
No

### How was this patch tested?

- vLLM version: v0.11.0rc3
- vLLM main: https://github.com/vllm-project/vllm/commit/v0.11.0

### CODE REVIEW

File: vllm_ascend/attention/mla_v1.py [hunk 1]
```diff
@@ -742,9 +742,8 @@ def _forward_prefill(
                                    k_rope=k_pe,
                                    value=value,
                                    mask=self.prefill_mask,
-                                   seqlen=torch.tensor(
-                                       attn_metadata.prefill.query_lens,
-                                       dtype=torch.int32),
+                                   seqlen=attn_metadata.prefill.query_lens.to(
+                                       torch.int32),
```
Comment (whx-sjtu) [inline vllm_ascend/attention/mla_v1.py]: Perform this `to(torch.int32)` operation also in `build` of attention metadata for performance consideration.

File: vllm_ascend/attention/mla_v1.py [hunk 2]
```diff
@@ -742,9 +742,7 @@ def _forward_prefill(
                                    k_rope=k_pe,
                                    value=value,
                                    mask=self.prefill_mask,
-                                   seqlen=torch.tensor(
-                                       attn_metadata.prefill.query_lens,
-                                       dtype=torch.int32),
+                                   seqlen=attn_metadata.prefill.query_lens.to(torch.int32),
```
[the remark this answers is not in the document]
Comment (whx-sjtu) [reply vllm_ascend/attention/mla_v1.py]: Also, don't forget to change the data type definition of `query_lens` for readablity. It should be torch.Tensor instead of List[int]

File: vllm_ascend/attention/mla_v1.py (hunk 1 above)
Comment (SunnyLee151064) [reply vllm_ascend/attention/mla_v1.py]: Modified yet

**Comment by yiz-liu** [comment]
@zzzzwwjj Please take a look at this, also related to attention metadata.
